### DISCUSSION TITLE
Would it be possible to expose TRANSIENT and FTCS MARKS variables to have profile override of OhMyPoshKeyHandler events?

### DISCUSSION THREAD

**Discussion by JayLCypher** [JanDeDobbeleer/oh-my-posh#4932] [Ideas]
Hello!

I've become addicted to a feature in my Linux distro terminal where I can just write paths directly without needing to cd everywhere.
Sorely missing when zooming in my Terminal on Windows.

I notice you've set up a few OhMyPoshKeyHandlers, however I don't see a way to safely override them keeping their behavior. The ::TRANSIENT:: and ::FTCS_MARKS:: variables are apparently dynamically overwritten on initialization? And there doesn't seem a way to access those variables from outside the omp.ps1 script (at least to my limited knowledge).
So I can only pick one of the Enter KeyHandlers.

Maybe I've been wasting time by trying, still don't fully understand what the event handlers does, but exposing some way to extend the handlers (if they are useful) would be appreciated.

Thanks for any consideration.

**Comment by JanDeDobbeleer** [comment]
@JayLCypher the handlers enable functionality and aren't meant to extend (at least not in the current design). You can actually override the current handler with whatever logic you want to add. Initialise oh-my-posh, then create your own handler based on the one we defined. The `Enter` handler is only used when you have `shell_integration` or `transient` prompt enabled, so if you're not using that you can override with anything you want, otherwise you'd need to extend the current behavior:

```powershell
Set-PSReadLineKeyHandler -Key Enter  -BriefDescription 'OhMyPoshEnterKeyHandler' -ScriptBlock {
    $executingCommand = $false
    try {
        $parseErrors = $null
        [Microsoft.PowerShell.PSConsoleReadLine]::GetBufferState([ref]$null, [ref]$null, [ref]$parseErrors, [ref]$null)
        $executingCommand = $parseErrors.Count -eq 0
    }
    finally {}

    [Microsoft.PowerShell.PSConsoleReadLine]::AcceptLine()

    # Write FTCS_COMMAND_EXECUTED after accepting the input - it should still happen before execution
    if ($executingCommand) {
        Write-Host "$([char]0x1b)]133;C`a" -NoNewline
    }
}
```

**Reply by JayLCypher** [reply]
Yeah, I had to reason about it since I noticed the CTRL-C handler was only registered in one if statement (thereby I chose that version of the handler).

Less about "extension" and more about a way to add some (unrelated?) logic before or after the handler.
Since there are two different handler versions running at initialization (let's call it "compile-time") checking against values that were replaced.

Since what I was trying to do was around the AcceptLine function, maybe that would be a point to have available an overloadable custom "acceptlinehandler" kind of deal. Similar to ValidateAndAcceptLine which calls to the CommandValidationHandler in PSConsoleReadLineOption.


In any case, I hacked a solution similar to your reply :)
Maybe it's something for the future.

Thanks for replying!
